packages/playwright/src/reporters/base.ts

```diff
@@ -489,11 +489,34 @@ export function stripAnsiEscapes(str: string): string {
   return str.replace(ansiRegex, '');
 }
 
+function characterWidth(c: string) {
+  return /[\u4e00-\u9fff]/.test(c) ? 2 : 1;
+}
+
+function stringWidth(v: string) {
+  let width = 0;
+  for (let i = 0; i < v.length; ++i)
+    width += characterWidth(v[i]);
+  return width;
+}
+
+function suffixOfWidth(v: string, width: number) {
+  let i = v.length - 1;
+  for (; i > 0; --i) {
+    const c = v[i];
```

Comment:
Same, this code can split surrogate pairs because it decrements the index i without considering whether v[i] is a full character or part of a surrogate pair